### ISSUE TITLE
Investigate why Weka mounting hangs often, especially in Nightly FD

### ISSUE THREAD

**Issue by tt-rkim** [tenstorrent/tt-metal#10944]
## Initial idea

Add retry steps to weka mount in needed pipelines to help with stability while we figure out how to better connect to Weka / use proper mounts

Started w/ FD nightly first on branch. If that worked, move onto perf and demos.

But that didn't - something is clearly wrong.

## Investigation

Looks like we can't access anything in MLPerf on certain VMs. It seems to be a problem with newer VMs

`umount` hangs without a clear reason. Can't even `lsof` it.

**Comment by tt-rkim** [comment]
What the hell? https://github.com/tenstorrent/tt-metal/actions/runs/10293588208/job/28516423230

Looks like it actually mounted after 2.5hr on this VM... damn...

**Comment by tt-rkim** [comment]
After some investigation, it looks like MLPerf hangs / refuses to unmount on certain newly-created VMs. This seems to happen on VMs created after mid-June 2024.

For now, to unblock, I will make some changes to our internal fleet management tools so that we can label all VMs created before this date to have a custom Weka tag that we can target for pipelines that rely on MLPerf.

**Comment by tt-rkim** [comment]
This has been done and working on a branch for #10395 . Lowering to P2.

**Comment by tt-rkim** [comment]
We're now targeting machines with label `mount-cloud-weka`, which is essentially any VM created before mid-June.

@ttmchiou @TT-billteng

**Comment by tt-rkim** [comment]
To recap so we don't lose context in future when we're looking at this again:

We noticed that certain jobs that required Weka (`mnt-MLPerf.mount systemctl service`) were not able to mount Weka within the job. It would hang and timeout.

At first, there didn't seem to be rhyme or reason for what was happening. We never had this issue before. However, I personally began to notice that certain machines seem to mount Weka okay, and that CI machines with higher VM numbers in their name (and likely therefore machines created more recently) seemed to have this problem. Older machines didn't seem to have this problem.

Logging into some newer machines, I saw that `umount` and `lsof` were hanging on `/mnt/MLPerf`. However, machines created before mid June 2024 don't seem to have this problem. `journalctl` also had no useful logs, whatsoever, as far as I could tell.

The current workaround to is to funnel jobs that require Weka to machines that are older than this date using a special label. We will investigate what's going on with the Weka service on newer machines later.
